tests/runtime/model_management/test_model_manager.py

```diff
@@ -1054,3 +1055,131 @@ def test_lazy_load_handles_temporary_errors():
                 assert manager._lazy_sync_timer is None
                 model = manager.retrieve_model(model_name)
                 assert model
+
+
+def test_lazy_load_true_local_models_dir_valid():
+    """When lazy_load_local_models is True and local_models_dir exists.
+    Check that the local_models_dir is pointing to the correct location
+    """
+
+    with TemporaryDirectory() as cache_dir:
+
+        ModelManager._ModelManager__instance = None
+        with temp_config(
+            {
+                "runtime": {
+                    "local_models_dir": cache_dir,
+                    "lazy_load_local_models": True,
+                }
+            },
+            merge_strategy="merge",
+        ):
+            MODEL_MANAGER = ModelManager()
+            assert len(MODEL_MANAGER.loaded_models) == 0
+            assert MODEL_MANAGER._local_models_dir == cache_dir
+
+
+def test_lazy_load_true_local_models_dir_invalid():
+    """When lazy_load_local_models is True and local_models_dir does not exist.
+    Raise ValueError with an appropriate message
+    """
+
+    with TemporaryDirectory() as cache_dir:
+
+        with pytest.raises(
+            ValueError,
+            match=(
+                "runtime.local_models_dir must be a valid path"
+                " if set with runtime.lazy_load_local_models. "
+                "Provided path: invalid"
+            ),
+        ) as exec_info:
+
+            ModelManager._ModelManager__instance = None
+            with temp_config(
+                {
+                    "runtime": {
+                        "local_models_dir": "invalid",
+                        "lazy_load_local_models": True,
+                    }
+                },
+                merge_strategy="merge",
+            ):
+                MODEL_MANAGER = ModelManager()
+
+        assert "runtime.local_models_dir must be a valid path" in str(exec_info.value)
```

Comment:
any reason for duplicating the `match` value as well as asserting here?

Reply:
I may have misunderstood, but I thought this was what you wanted based on previous feedback.
https://github.com/caikit/caikit/pull/591#discussion_r1416161055

Reply:
Yes but from what I understand but doesn't the `match` value at the beginning of the test do that already?